Automattic/wp-calypso
# Stepper: Redirect user to the login page using the correct language

## Proposed Changes

* Redirect the user to sign the flow using the lang defined on the flow path (E.g `/setup/migration-signup/de` )
* Fix the redirect to `/setup/:flowName` to the first step to include the lang
* Quickly redirect the user to the `site-setup` flow when there is no flow on the URL.
* Centralize the logic to get the location on the `'ui.language.localeSlug'
* Improve logic to determine the flow step and locale based on the URL. 
* Update flows to use useFlowLocale instead of useLocale


> [!NOTE]  
> There is a possible race condition so you might see the steps rendering in English before rendering the selected language. It will be fixed as part of other PR.


## Why are these changes being made?
Nowadays when the user accesses/setup/:flowName/:lang we remove the lang part when we redirect the user to the first flow step. 

## Testing Instructions



* Using a not logged-in browser, access `/setup/migration-signup/de`
* Check if the signup page is in German
* Use a random email on the email sign-up to create a new account
* Check if you see the steps in German.


## Pre-merge Checklist
